### DISCUSSION TITLE
Quality of parental control feature

### DISCUSSION THREAD

**Discussion by timkgh** [AdguardTeam/AdGuardHome#3675] [Q&A]
I've run into 2 cases of false positives. I understand that false positives happen occasionally and we fix them by reporting them. What's concerning is that these 2 broke before, got fixed, then they broke again. Do you have any processes in place to make sure that sites don't re-appear on block lists when they shouldn't?

**Comment by timkgh** [comment]
I keep running into false positives that should not be blocked. I report them but the issues just sit there unactioned. I agree that some of these can be tricky, they point to buckets in S3 or B2 or to software repositories that _may_ also host adult content, but they also have legitimate content, e.g. backups that I cannot get to. DNS blocking is just not the right way to do it in this case, it is too wide. Content filtering that can inspect the URL is needed.

**Comment by timkgh** [comment]
@ameshkov This is becoming a problem, I keep running into domains of the form `*.s3.us-west-2.amazonaws.com` that keep getting blocked as "adult website" when these buckets do not really have any such content. I know I can allow list them, but do you think AGH should reconsider how it does some of the DNS blocking when it comes to services such as S3 and B2?

I've also run into multiple cases where a domain is _incorrectly_ blocked by AGH with parental control enabled, but when I resolve them via `94.140.14.15` (family protection) they are not blocked. Doesn't AGH use `94.140.14.15` to decide what to block for parental control? It does not seem to work.

**Reply by ameshkov** [reply]
AGH does use `94.140.14.15` indeed. Could you please provide some real-life examples?

**Reply by timkgh** [reply]
Any way to send you a private email?

**Reply by ameshkov** [reply]
yep, `am@adguard.com`

**Reply by timkgh** [reply]
Sent. Referenced this discussion # in the subject.
